File: VB/Step 3 - creating view 2 (PersonsViewModel and PersonsView)/ViewModels/PersonViewModel.vb
Imports System
Imports DX_MVVM.Models

Namespace DX_MVVM.ViewModels
    Public Class PersonViewModel
        Inherits ViewModelBase

        Public Property FirstName() As String
            Get
                Return Person.FirstName
            End Get
            Set(ByVal value As String)
                If Person.FirstName = value Then
                    Return
                End If
                Person.FirstName = value
                OnPropertyChanged("FirstName")
            End Set
        End Property
        Public Property LastName() As String
            Get
                Return Person.LastName
            End Get
            Set(ByVal value As String)
                If Person.LastName = value Then
                    Return
                End If
                Person.LastName = value
                OnPropertyChanged("LastName")
            End Set
        End Property
        Public Property Photo() As Uri
            Get
                Return Person.Photo
            End Get
            Set(ByVal value As Uri)
                If Person.Photo = value Then
                    Return
                End If
                Person.Photo = value
                OnPropertyChanged("Photo")
            End Set
        End Property
        Public Property Email() As String
            Get
                Return Person.Email
            End Get
            Set(ByVal value As String)
                If Person.Email = value Then
                    Return
                End If
                Person.Email = value
                OnPropertyChanged("Email")
            End Set
        End Property
        Public Property IsReadOnly() As Boolean
            Get
                Return isReadOnly_Renamed
            End Get
            Set(ByVal value As Boolean)
                If isReadOnly_Renamed = value Then
                    Return
                End If
                isReadOnly_Renamed = value
                OnPropertyChanged("IsReadOnly")
            End Set
        End Property
        Private privatePerson As Person
        Public Property Person() As Person
            Get
                Return privatePerson
            End Get
            Private Set(ByVal value As Person)
                privatePerson = value
            End Set
        End Property

        Public Sub New(ByVal person As Person)
            Me.Person = person
        End Sub


        Private isReadOnly_Renamed As Boolean = False
    End Class
End Namespace

File: CS/Step 4 - building views/ViewModels/PersonViewModel.cs
using System;
using DX_MVVM.Models;

namespace DX_MVVM.ViewModels {
    public class PersonViewModel : ViewModelBase {
        public String FirstName {
            get { return Person.FirstName; }
            set {
                if(Person.FirstName == value) return;
                Person.FirstName = value;
                OnPropertyChanged("FirstName");
            }
        }
        public String LastName {
            get { return Person.LastName; }
            set {
                if(Person.LastName == value) return;
                Person.LastName = value;
                OnPropertyChanged("LastName");
            }
        }
        public Uri Photo {
            get { return Person.Photo; }
            set {
                if(Person.Photo == value) return;
                Person.Photo = value;
                OnPropertyChanged("Photo");
            }
        }
        public String Email {
            get { return Person.Email; }
            set {
                if(Person.Email == value) return;
                Person.Email = value;
                OnPropertyChanged("Email");
            }
        }
        public bool IsReadOnly {
            get { return isReadOnly; }
            set {
                if(isReadOnly == value) return;
                isReadOnly = value;
                OnPropertyChanged("IsReadOnly");
            }
        }
        public Person Person { get; private set; }

        public PersonViewModel(Person person) {
            Person = person;
        }

        bool isReadOnly = false;
    }
}
